railties/lib/rails/generators/rails/app/templates/config/initializers/content_security_policies.rb.tt

```diff
@@ -0,0 +1,21 @@
+# Define application-wide content security policies
+# For further information see the following documentation
+# https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
+
+Rails.application.config.content_security_policies.build do
+  policy :application do |p|
+    p.default_src :self, :https
+    p.font_src    :self, :https, :data
+    p.img_src     :self, :https, :data
+    p.object_src  :none
+    p.script_src  :self, :https
+    p.style_src   :self, :https, :unsafe_inline
+  end
+
+  # Report CSP violations to the specified URI
+  # For further information see the following documentation:
+  # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only
+  policy :report_only, using: :application do |p|
+    p.report_only!(uri: "/csp-violation-report-endpoint")
+  end
+end
```

Comment:
Not sure about this policy collection API for central definition of multiple policies. I think in practice we'll have a single base policy that's extended in controller subclasses. If there's a set of controllers that use a stricter policy, I'd probably start by mixing in a concern that sets the strict policy.

Reply:
Agreed - a single base policy with overrides either in concern or a common parent controller class makes the implementation simpler and doesn't lose any flexibility.